src/cueUtilities.js

```diff
@@ -129,6 +129,32 @@ module.exports = function (params, cy, api) {
             x : x,
             y : y
           };
+        } else if (options().expandCollapseCuePosition === 'top-right') {
+          var offset = 1;
+          var size = cy.zoom() < 1 ? rectSize / (2*cy.zoom()) : rectSize / 2;
+
+          var x = node.position('x') + node.width() / 2 + parseFloat(node.css('padding-left'))
+                  - parseFloat(node.css('border-width')) - size - offset;
+          var y = node.position('y') - node.height() / 2 - parseFloat(node.css('padding-top'))
+                  + parseFloat(node.css('border-width')) + size + offset;
+
+          cueCenter = {
+            x : x,
+            y : y
+          };
+        } else if (options().expandCollapseCuePosition === 'top-center') {
+          const x = node.position('x');
+          let y;
+          if (cy.expandCollapse('get').isExpandable(node)) {
+              y = node.position('y');
+          } else {
+              y = node.position('y') - 12 - node.height() / 2;
```

Comment:
Ack! I know that 12 is from my original code, but it was bad of me to do that, and it definitely shouldn't be here in the add-on. Try to calculate the y the same way the other position options do.

There's a lot of similarity in the code for these three position options, so see if you can consolidate some of it. Maybe have an initial `if` checking if the position begins with 'top' and calculate the y, offset, size etc for all three settings.